### ISSUE TITLE
Assignment 6 huge PCA Matrixes

### ISSUE THREAD

**Issue by baclemen** [eth-igl/GP2021-Assignments#93]
Calculating the covariance Matrix for the data within example2 (/aligned_faces_example/example2/) is not really possible since it needs about 100 GB of memory.

Is there a good other way to compute the eigenfaces without calculating A * A^T?

**Comment by wsh312** [comment]
You are right, please use a lower resolution of mesh, like those in example1
For large matrix decomposition, maybe you can try other libraries like the Spectra library (https://spectralib.org).
If the non-rigid deformation is done on the high-resolution mesh, one needs to implement something to transfer all the high-res to low-res meshes.
So I suggest communicating and define a good resolution to work on.

**Comment by FloorVerhoeven** [comment]
@baclemen 
As an additional remark (as I'm not sure if you're doing this already): if the number of data points is much smaller than the dimension of each datapoint, you can instead compute the eigenfaces via the eigenvectors of A^T*A (which would be much smaller). Please see [this link,](https://www.face-rec.org/algorithms/PCA/jcn.pdf) pages 74 and 75 for a more detailed explanation.

**Comment by wsh312** [comment]
Agon suggested this link that well explains these the two solutions: https://math.stackexchange.com/questions/3869/what-is-the-intuitive-relationship-between-svd-and-pca 

Both solutions are valid.
